include/Metabox/Posts/CategoryToTag.php

```diff
@@ -30,19 +30,40 @@ public function render() {
             <table class="optiontable">
                 <tr>
                     <td scope="row">
-						<?php
-						wp_dropdown_categories( array(
-							'name'         => 'smbm_mct_cat',
-							'show_count'   => true,
-							'hierarchical' => true,
-							'orderby'      => 'NAME',
-							'hide_empty'   => false,
-						) );
-						?>
+                    	<?php
+                        $categories = get_categories( array(
+                                'hide_empty' => false,
+                            )
+                        );
+
+                        if( count($categories) > 50 ){?>
+                            <select class="select2Ajax" name="smbm_mct_cat" data-term="category" data-placeholder="<?php _e( 'Select Category', 'bulk-move' ); ?>" style="width:300px">
+                            </select>
+                        <?php }else{?>
+                            <select class="select2" name="smbm_mct_cat" data-placeholder="<?php _e( 'Select Category', 'bulk-move' ); ?>">
+                            <?php foreach ( $categories as $category ) { ?>
+                                <option value="<?php echo $category->cat_ID; ?>"><?php echo $category->cat_name, ' (', $category->count, ' ', __( 'Posts', 'bulk-move' ), ')'; ?></option>
+                            <?php } ?>
+                            </select>
+                        <?php }
+                        ?>
                         ==>
                     </td>
                     <td scope="row">
-						<?php $this->render_tags_dropdown( 'smbm_mct_tag', $tags ); ?>
+                    	<?php
+                        $tags = get_tags();
```

Comment:
Move the code that retrieves the tags and renders it into a seperate helper function to the `BM_Metabox_PostBase` class.